src/snowflake/snowpark/session.py

```diff
@@ -1472,6 +1472,12 @@ def create_dataframe(
             not installed_pandas
             or (installed_pandas and not isinstance(data, pandas.DataFrame))
         ):
+            if isinstance(data, pandas.DataFrame):
```

Comment:
I'm wondering, if pandas is not installed, how can user pass in a pandas dataframe?

Reply:
we can install pandas separately right?

Reply:
I think this is because in the connector, `installed_pandas` is only True when both `pandas` and `pyarrow` are installed, but later `write_pandas` will fail if `pyarrow` is not installed anyway...

Reply:
I believe it is helpful here to indicate the user that the python connector is not installed as expected here.

Reply:
I see what you mean. Let's move separate the two if cases then, one for `not installed_pandas` and one for `not isinstance(data, (list, tuple, pandas.DataFrame))`.

Reply:
good point! Can you take another look?